File: Scripts/2024/find_range.m
function [J] = find_range(J)
        % Compute the range of the jacobian
        Range = orth(sym(J));

        % Display the range
        disp('range of the jacobian');
        disp('warning! this result might be wrong, please check by hand if possible')
        R=simplify(Range);
        [n,m]=size(R);
        for i=1:m
            [numerator, denominator] = numden(R(:,i));
            fprintf('%d basis',i)
            display(simplify(numerator,'Steps',50));
        end
end

File: Scripts/2025/python_scripts/robobox/find_range.py
import sympy as sp

def find_range(M: sp.Matrix) -> list[sp.Matrix]:
    """
    Computes and displays the range (image) of a matrix

    @param J: The Jacobian matrix
    """
    range_space = M.columnspace()

    print("Warning! This result might be wrong, please check by hand if possible.")
    print("Range of the Jacobian:")
    for i, vec in enumerate(range_space, start=1):
        simplified_vec = sp.simplify(vec)
        print(f"Basis vector n°{i}:")
        sp.pprint(simplified_vec)